benchmarks/profiler/profile_sla.py

```diff
@@ -173,410 +190,335 @@ async def run_profile(args):
+                        namespace=args.namespace,
+                        base_log_dir=work_dir,
+                        model_name=model_name,
+                        service_name=args.service_name,
+                        frontend_port=frontend_port,
+                        deployment_name=prefill_config["metadata"]["name"],
+                    )
+                    logger.info(
+                        f"Created client with service_name: {client.service_name}"
+                    )
+                    deployment_clients.append(client)  # Track for cleanup
+                    await client.create_deployment(prefill_config_fn)
+                    logger.info("Waiting for deployment to be ready...")
+                    await client.wait_for_deployment_ready()
+                    logger.info("Deployment is ready")
+
+                    logger.info("Getting deployment logs...")
+                    await client.get_deployment_logs()
+                    logger.info(
+                        f"Logs have been saved to {client.base_log_dir / client.deployment_name}"
+                    )
 
-                # run ai-perf
-                base_url = client.get_service_url()
-                ai_perf_artifact_dir = f"{work_dir}/aiperf_isl{args.isl}"
-                aiperf_result = benchmark_prefill(
-                    args.isl,
-                    ai_perf_artifact_dir,
-                    model_name,
-                    model_name,
-                    base_url=base_url,
-                )
-                if aiperf_result is not None:
-                    ttft = aiperf_result["time_to_first_token"]["avg"]
+                    # run ai-perf
+                    base_url = client.get_service_url()
+                    ai_perf_artifact_dir = f"{work_dir}/aiperf_isl{args.isl}"
+                    aiperf_result = benchmark_prefill(
+                        args.isl,
+                        ai_perf_artifact_dir,
+                        model_name,
+                        model_name,
+                        base_url=base_url,
+                    )
+                    if aiperf_result is not None:
+                        ttft = aiperf_result["time_to_first_token"]["avg"]
 
-                logger.info("Cleaning up deployment...")
-                await client.delete_deployment()
-                deployment_clients.remove(client)
-                logger.info("Deployment deleted")
+                    logger.info("Cleaning up deployment...")
+                    await client.delete_deployment()
+                    deployment_clients.remove(client)
+                    logger.info("Deployment deleted")
 
-            if ttft is not None:
-                prefill_num_gpus.append(num_gpus)
-                prefill_ttft.append(ttft)
-                prefill_thpt_per_gpu.append(args.isl / ttft / num_gpus * 1000)
+                if ttft is not None:
+                    prefill_data.num_gpus.append(num_gpus)
```

Comment:
Ah, when I said enum, I meant that the prefill data for one mapping would be an enum; then you could have a list of prefill_data and append a singular value to the list of a new prefill_data object instead of having like 6 append lines when each is guaranteed to be the same length + same indexing